app/controllers/users/registrations_controller.rb

```diff
@@ -1,22 +1,22 @@
 # frozen_string_literal: true
-
 class Users::RegistrationsController < Devise::RegistrationsController
-
   include RackSessionsFix
   respond_to :json
 
-  private
-
-  def respond_with(current_user, _opts = {})
-    if resource.persisted?
-      render json: {
-        status: {code: 200, message: 'Signed up successfully.'},
-        data: UserSerializer.new(current_user).serializable_hash[:data][:attributes]
-      }
+  def create
+    contract = Contracts::Users::UserRegistrationContract.new
+    operation = contract.call(params.permit(user: {}).to_h)
+    if operation.success?
```

Comment:
The operation should be creating the user.  This logic shouldn't be in the controller.  The only thing the controller should be doing is permitting parameters, calling an operation, checking the result and rendering the response as appropriate.